core/src/main/java/org/infinispan/commands/write/BackupPutMapAckCommand.java

```diff
@@ -0,0 +1,97 @@
+package org.infinispan.commands.write;
+
+import org.infinispan.commands.CommandInvocationId;
+import org.infinispan.commands.remote.BaseRpcCommand;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.util.ByteString;
+import org.infinispan.util.concurrent.CommandAckCollector;
+import org.infinispan.util.concurrent.CompletableFutures;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A command that represents an acknowledge sent by a backup owner to the originator.
+ * <p>
+ * The acknowledge signals a successful execution of the the {@link PutMapCommand}. It contains the segments ids of
+ * the updated keys.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public class BackupPutMapAckCommand extends BaseRpcCommand {
+
+   public static final byte COMMAND_ID = 41;
+   private CommandInvocationId commandInvocationId;
+   private CommandAckCollector commandAckCollector;
+   private Collection<Integer> segments;
+   private int topologyId;
+
+   public BackupPutMapAckCommand() {
+      super(null);
+   }
+
+   public BackupPutMapAckCommand(ByteString cacheName) {
+      super(cacheName);
+   }
+
+   public BackupPutMapAckCommand(ByteString cacheName, CommandInvocationId commandInvocationId, Collection<Integer> segments, int topologyId) {
+      super(cacheName);
+      this.commandInvocationId = commandInvocationId;
+      this.segments = segments;
+      this.topologyId = topologyId;
+   }
+
+   @Override
+   public Object invoke() throws Throwable {
+      commandAckCollector.putMapBackupAck(commandInvocationId, getOrigin(), segments, topologyId);
+      return null;
```

Comment:
actually, ```perform()```, ```invoke()``` and ```invokeAsync()``` have a default implementation. It was kind of confusing which one to implement :)